src/utl/include/utl/prometheus/metrics_server.h

```diff
@@ -0,0 +1,58 @@
+// Copyright 2025 Google LLC
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file or at
+// https://developers.google.com/open-source/licenses/bsd
+
+#pragma once
+
+#include <atomic>
+#include <chrono>
+#include <fstream>
+#include <iostream>
+#include <memory>
+#include <string>
+#include <thread>
+
+#include "registry.h"
+#include "text_serializer.h"
+
+namespace utl {
+class Logger;
+}
+
+namespace utl {
+class PrometheusMetricsServer
+{
+  std::thread worker_thread_;

```

Comment:
The private section goes last https://google.github.io/styleguide/cppguide.html#Declaration_Order